ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/WrappingAuditAuthenticator.java

```diff
@@ -96,41 +98,59 @@ public AuthenticatedUser legacyAuthenticate(Map<String, String> credentials) thr
         return wrappedAuthenticator.legacyAuthenticate(credentials);
     }
 
+    @Override
+    public AuditSaslNegotiator createAuditedSaslNegotiator()
+    {
+        return new AuditingSaslNegotiator(wrappedAuthenticator.createAuditedSaslNegotiator());
+    }
+
+    @Override
+    public Set<IRoleManager.Option> supportedOptions()
```

Comment:
The WrappingAuditAuthenticator should add OPTIONS to the set provided by the wrappedAuthenticator, before returning it to the calling method. For the scenarios where the wrappedAuthenticator isn't our own AuditPasswordAuthenticator.

Reply:
Should we split up the interface? One for IAuditAuthenticator and ISomethingGoodName for providing the options? And WrappingAuditAuthenticator can implement only the latter.